File: assets/rt/Fallible.java
package rt;

import base.Action_1;
import base.MF_2;

public interface Fallible extends Action_1 {
  @Override default Object $exclamation$mut() {
    return Action_1.$exclamation$mut$fun(this);
  }
  @Override default Action_1 mapInfo$mut(MF_2 f_m$) {
    return Action_1.mapInfo$mut$fun(f_m$, this);
  }
  @Override default Action_1 map$mut(MF_2 f_m$) {
    return Action_1.map$mut$fun(f_m$, this);
  }
  @Override default Action_1 andThen$mut(MF_2 f_m$) {
    return Action_1.andThen$mut$fun(f_m$, this);
  }
  @Override default base.Opt_1 ok$mut() { return Action_1.ok$mut$fun(this); }
  @Override default base.Opt_1 info$mut() { return Action_1.info$mut$fun(this); }
}

File: assets/rt-js/Fallible.js
import { base$$Action_1 } from "../base/Action_1.js";

export class Fallible {
  constructor(runFn) {
    // store the function that will act as run$mut$1
    this.run$mut$1 = runFn;
  }

  $exclamation$mut$0() {
    return base$$Action_1.$exclamation$mut$1$fun(this);
  }

  mapInfo$mut$1(f_m$) {
    return base$$Action_1.mapInfo$mut$2$fun(f_m$, this);
  }

  map$mut$1(f_m$) {
    return base$$Action_1.map$mut$2$fun(f_m$, this);
  }

  andThen$mut$1(f_m$) {
    return base$$Action_1.andThen$mut$2$fun(f_m$, this);
  }

  ok$mut$0() {
    return base$$Action_1.ok$mut$1$fun(this);
  }

  info$mut$0() {
    return base$$Action_1.info$mut$1$fun(this);
  }
}
